org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/actions/RulerToggleHitBreakpointActionDelegate.java

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.internal.debug.ui.actions;
+
+import org.eclipse.jdt.debug.core.IJavaLineBreakpoint;
+import org.eclipse.jdt.internal.debug.ui.JDIDebugUIPlugin;
+import org.eclipse.jface.dialogs.IInputValidator;
+import org.eclipse.jface.dialogs.InputDialog;
+import org.eclipse.jface.text.ITextSelection;
+import org.eclipse.jface.window.Window;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.ui.texteditor.ITextEditor;
+
+public class RulerToggleHitBreakpointActionDelegate extends AbstractRulerToggleBreakpointActionDelegate {
+
+	@Override
+	protected boolean doWork(ITextEditor editor, ITextSelection selection) {
+		IJavaLineBreakpoint jlp = ToggleBreakpointAdapter.findExistingBreakpoint(currentEditor, selection);
+		if (jlp == null) {
+			hitCountDialog();
+			if (BreakpointToggleUtils.getHitCount() < 2) {
```

Comment:
Why is this condition `< 2`? This looks strange, also the check for the values checks for `< 1`

Reply:
Fixed